### DISCUSSION TITLE
Mac Development

### DISCUSSION THREAD

**Discussion by decuser** [tsujan/FeatherPad#615] [General]
Is featherpad currently buildable on Mac? Are there any dev setup notes for mac?

**Comment by tsujan** [comment]
> Is featherpad currently buildable on Mac?

Yes, @shlyakpavel added the support, to both qmake and cmake compilations.

> Are there any dev setup notes for macOS?

As I (FeatherPad's author) don't have macOS, I can't tell. I use it on Linux. However, I'd appreciate it if the needed info is added to `INSTALL`.

**Comment by shlyakpavel** [comment]
> Is featherpad currently buildable on Mac? 

Yes
> Are there any dev setup notes for Mac?

You need cmake/qmake, Qt from http://qt.io and hunspell build from https://github.com/hunspell/hunspell (not from homebrew!!)
When everything is as on linux. You can also use Qt Creator (just build/run).

**Reply by shlyakpavel** [reply]
Important: you need Qt6 to build it on arm macs

**Comment by decuser** [comment]
@shlyakpavel: I downloaded the source, QT creator, and installed hunspell. When I went to build it, it errored out:

`Project ERROR: hunspell development package not found
make: *** [sub-featherpad-qmake_all] Error 3
08:59:50: The process "/usr/bin/make" exited with code 2.
`

After installing hunspell, I ran 'sudo update_dyld_shared_cache', any idea how I get QT Creator to know about hunspell?

**Reply by shlyakpavel** [reply]
install pkg-config (from homebrew, for example)
Did you try to open the project as cmake or qmake in Qt Creator?

**Comment by decuser** [comment]
I installed pkg-config from macports. I just opened the .pro file. It didn't ask which (cmake vs qmake). However, it talks about qmake in the compiler output window.

**Reply by shlyakpavel** [reply]
Open cmakelists.txt in Qt creator as a project

**Comment by decuser** [comment]
OK. Well at least the error is more descriptive:

`CMake Error at /opt/local/share/cmake-3.21/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find HUNSPELL: Found unsuitable version "", but required is at
  least "1.6" (found /usr/local/lib/libhunspell-1.6.dylib)
`

I tried it with latest hunspell and this version 1.6.2, looks like hunspell isn't reporting it's version number. I built hunspell from source as per instruction above.

**Comment by shlyakpavel** [comment]
I had the same problem. Check if your hunspell installation is correctly found by PkgConfig. It works perfectly if PkgConfig is installed by homebrew, but it may differ with macports paths. Also double check that all libraries and compilers are for the same architecture (x86-64 or arm64).

**Reply by decuser** [reply]
How do I check the PkgConfig installation? I'm sure it's different, but I'm new to QT development and C/Qmake. I know that with Brew, it's in /usr/local and with Port, it's in /opt/local, but beyond that, I'm not sure. Everything's x86-64.

**Reply by shlyakpavel** [reply]
Manually find hunspell.pc file path and manually add its path to PKG_CONFIG_PATH environment variable (in Qt creator in build environment section. Homebrew also uses /opt on arm64, so it's not a big difference. Also, ensure pkg-config itself is in path and you can run it from terminal.

**Reply by shlyakpavel** [reply]
We can also have a skype/zoom/jitsi call and try to resolve this issue together. Mojave is already out of date and I cannot get the same environment you have here, even on a VM

**Reply by decuser** [reply]
I tried adding PKG_CONFIG_PATH /lib/pkgconfig/hunspell.pc, but no joy. A Zoom call sounds great. When's good for you and what timezone?

**Reply by shlyakpavel** [reply]
/lib/pkgconfig/hunspell.pc
Do you really have this file in /lib? O_o
You should add /lib/pkgconfig/ as PKG_CONFIG_PATH then. I will be happy to speak with you 14th December at 19:00 Moscow time. I can post the zoom link here in this thread.

**Reply by decuser** [reply]
Ah, my mistake it's /usr/lib/pkgconfig. Either way, no dice. If I read the timezone diff, we're talking about 16 hours from now? Sure.

**Reply by shlyakpavel** [reply]
> Ah, my mistake it's /usr/lib/pkgconfig. Either way, no dice. If I read the timezone diff, we're talking about 16 hours from now? Sure.

It's 00:47 13th December now
It will be 00:47 14th December in 24 hours from now
It will be 18:47 in 24+18=42 hours from now. 
It will be 19:00 in 42 hours and 13 minutes from now

**Reply by decuser** [reply]
NM! I figured it out. Just a cut and paste issue. The path was /usr/local/lib/pkgconfig. Thanks for the tips, it's building fine now.

**Reply by shlyakpavel** [reply]
Fine. There is no need to have a call then. Feel free to write here if you have additional questions

**Reply by shlyakpavel** [reply]
The issue may be that macports pkgconfig has different defaults for paths. I don't think I need to dig it deeper as anyone can now find this thread and set the environment variable. For those who come here searching "hunspell not found" : check your library arches twice on m1, it's really easy to mess them.

**Comment by shlyakpavel** [comment]
Is it m1 Mac?

**Reply by decuser** [reply]
No, it's an old MacBook Pro i7, running Mojave.
